jdaviz/configs/cubeviz/plugins/spectral_extraction/tests/test_spectral_extraction.py

```diff
@@ -569,6 +570,17 @@ def test_default_spectral_extraction(cubeviz_helper, spectrum1d_cube_fluxunit_jy
         extracted_spectra[0].flux, extracted_spectra[1].flux
     )
 
+    # Also test unit conversion
+    uc = cubeviz_helper.plugins["Unit Conversion"]
+    assert uc.flux_unit == "Jy"
+    uc.flux_unit = "MJy"
+    spec_extr_plugin = cubeviz_helper.plugins['Spectral Extraction']
+    collapsed = spec_extr_plugin.extract()
+    # Actual values not in display unit but should not affect display unit.
+    assert collapsed.flux.unit == u.Jy
+    assert uc.flux_unit.selected == "MJy"
+    assert cubeviz_helper.app._get_display_unit('spectral_y') == "MJy"
```

Comment:
This actually does not fail on `main` even though I see the bug when I re-extract the one spectrum manually. Not sure how to make this test actually catch that bug. 🤷‍♀️

Reply:
the last two lines are effectively doing the same thing.  If the viewer is unsyncing from the unit conversion plugin, then we'll need to check the actual units on the viewer (`viewer.state.x_display_unit` or something like that).

Reply:
But I did confirm manually that this patch should fix the original problem Clare reported.

Reply:
Oh, oops. I didn't see there is a subset in  that test. I moved the test code to a new function and it fails on main but passes here. I will push new commit here.